turbo/consumers.py

```diff
@@ -47,16 +40,25 @@ def notify(self, event, *args, **kwargs):
             })
 
     def receive_json(self, content, **kwargs):
-        model_label = content.get("model")
-        pk = content.get("pk")
+        signer = Signer()
         request_id = content.get("request_id")
+        message_type = content.get("type")
+        if message_type == "subscribe":
+            try:
+                channel_name = signer.unsign(content.get("signed_channel_name", ""))
+            except BadSignature:
+                print("Signature has been tampered with!")
+                return
```

Comment:
I might fail loudly here. As an app operator I'd prefer to get an actual exception if someone is trying to break my security. Maybe instead of a `print`, rethrowing a `BadSigniture` with the same message would help users see they are being attacked?

Reply:
Yeah, I think we probably need to think about a comprehensive strategy to error handling. In decoupled SPA apps, I'd return an error over JSON. But it's weird here since it's a fully server-rendered app, yet in this one case we're sending a request from the frontend, even if it's triggered by HTML markup that the server filled in. A server-side error does make sense when I think about it, so for now, I'll change this to rethrow a custom `TurboStreamsException` in any place that might throw an error!

Reply:
Nice. Yes, error handling is an interesting challenge in this context. In this case the client has done something they know is wrong, or it's a programming error, so I'm not sure a client error is effective here. I could see us sending an error to the client for other issues though.

Reply:
I wonder if this is something where it makes sense to add something like Prometheus or other metric trackers to track such events?

Reply:
We could certainly add some hooks for people to get metrics from, but I'm hesitant to incorporate any solution specifically. I'd say we wait until someone actually requests it though.

Throwing an exception is nice since every production app has some way to catch and surface errors. Not everyone handles logs or metrics though.